core/src/main/java/com/microsoft/applicationinsights/internal/profile/CdsRetryPolicy.java

```diff
@@ -0,0 +1,62 @@
+package com.microsoft.applicationinsights.internal.profile;
+
+/**
+ * Responsible for Cds Retry Policy configuration.
+ */
+public enum CdsRetryPolicy {
+    /**
+     * Cached instance to be reused across SDK for CDS Profile fetch calls.
+     */
+    INSTANCE;
+
+    /**
+     * Maximum number of instant retries to CDS to resolve ikey to AppId.
+     */
+    private int maxInstantRetries;
+
+    /**
+     * The interval in minutes for retry counters and pending tasks to be cleaned.
+     */
+    private long resetPeriodInMinutes;
+
+    public int getMaxInstantRetries() {
+        return maxInstantRetries;
+    }
+
+    public long getResetPeriodInMinutes() {
+        return resetPeriodInMinutes;
+    }
+
+    public void setMaxInstantRetries(int maxInstantRetries) {
+        if (maxInstantRetries < 1) {
+            throw new IllegalArgumentException("Number of CDS instant retries should be atleast 1");
```

Comment:
atleast -> at least

normally people would put "maxInstantRetries must be greater than or equal to 1"